lib/exq/node/server.ex

```diff
@@ -0,0 +1,103 @@
+defmodule Exq.Node.Server do
+  use GenServer
+  require Logger
+  alias Exq.Support.Config
+  alias Exq.Support.Time
+  alias Exq.Redis.JobStat
+  alias Exq.Support.Node
+
+  defmodule State do
+    defstruct [
+      :node,
+      :interval,
+      :namespace,
+      :redis,
+      :node_id,
+      :manager,
+      :workers_sup,
+      ping_count: 0
+    ]
+  end
+
+  def start_link(options) do
+    node_id = Keyword.get(options, :node_id, Config.node_identifier().node_id())
+
+    GenServer.start_link(
+      __MODULE__,
+      %State{
+        manager: Keyword.fetch!(options, :manager),
+        workers_sup: Keyword.fetch!(options, :workers_sup),
+        node_id: node_id,
+        node: build_node(node_id),
+        namespace: Keyword.fetch!(options, :namespace),
+        redis: Keyword.fetch!(options, :redis),
+        interval: 5000
+      },
+      []
+    )
+  end
+
+  def init(state) do
+    :ok = schedule_ping(state.interval)
+    {:ok, state}
+  end
+
+  def handle_info(
+        :ping,
+        %{
+          node: node,
+          namespace: namespace,
+          redis: redis,
+          manager: manager,
+          workers_sup: workers_sup
+        } = state
+      ) do
+    {:ok, queues} = Exq.subscriptions(manager)
+    busy = Exq.Worker.Supervisor.workers_count(workers_sup)
+    node = %{node | queues: queues, busy: busy, quiet: Enum.empty?(queues)}
+
+    :ok =
+      JobStat.node_ping(redis, namespace, node)
+      |> process_signal(state)
+
+    if Integer.mod(state.ping_count, 10) == 0 do
```

Comment:
What is 10 for?  Should be configured elsewhere or constant?

Reply:
We have to remove the dead nodes from the processes set, the mod 10 is just a optimization to avoid doing the cleanup check on each ping instead do it on every 10th ping. The number 10 itself comes from [sidekiq implementation](
https://github.com/mperham/sidekiq/blob/46f4c6bd9ee0b2f5502a1dc0d03414989ad65063/lib/sidekiq/scheduled.rb#L160). If you feel this should be configurable, I can move it to application config